### DISCUSSION TITLE
ResolversEnhanceMap

### DISCUSSION THREAD

**Discussion by mihir-rughani** [MichalLytek/typegraphql-prisma#153] [General]
Hi all,

Probably a stupid question but would anyone be able to tell me how you implement ResolversEnhanceMap, please?

I've created a file and added the code as per the example:

```
import {
  ResolversEnhanceMap,
  applyResolversEnhanceMap,
} from "@generated/type-graphql";
import { Authorized } from "type-graphql";

const resolversEnhanceMap: ResolversEnhanceMap = {
  Category: {
    createCategory: [Authorized(Role.ADMIN)],
  },
};
applyResolversEnhanceMap(resolversEnhanceMap);
```

but authChecker is not being called. Where should this code go? and how do I ensure it's being run?  

Many thanks in advance for your help

**Comment by MichalLytek** [comment]
It has to be inside `index.ts` - it can't be a separate file not imported anywhere. The code has to be in the import chain of your entrypoint, as the functions has to be called in runtime.

**Reply by mihir-rughani** [reply]
Thank you so much; I figured that was where it was supposed to go but wasn't sure. I want to say thank you for these excellent frameworks you've developed; both type-graphql and typegraphql-prisma are amazing. You're a legend!
